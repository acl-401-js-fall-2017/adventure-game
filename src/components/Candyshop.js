import React, { Component } from 'react';
import shopPic from '../pics/wonkaStore.jpg';
import PropTypes from 'prop-types';


export default class Candyshop extends Component {

  constructor() {
    super();
    this.state = {
      message: '',
    };
  }

  handleClick = (value, inv = '') => {
    const { handleInventory, inventory } = this.props;
    this.setState({
      message: value !== 'correct' ? 'Pick again' : 'You have got the golden ticket'
    },() =>  {
      if(inventory.indexOf('golden-ticket') === -1 && inv === 'golden-ticket') {
        handleInventory('golden-ticket');
      }
    });
    
  }

  render() {
    
    return (
      <section className="challenge">
        <p className="picture">
          <img  src={shopPic} useMap="#shopMap" alt="Candy shop"/>
          <map name="shopMap" id="shopMap">
            <area alt="grid"  data-value="wrong" shape="poly" coords="53,194,52,320,220,324,220,208" onClick={this.handleClick}/>
            <area alt="grid"  data-value="wrong" shape="poly" coords="61,342,57,415,216,416,221,345" onClick={this.handleClick}/>
            <area alt="grid"  data-value="wrong" shape="poly" coords="56,444,52,518,205,516,220,443" onClick={this.handleClick}/>
            <area alt="grid"  data-value="wrong" shape="poly" coords="262,208,261,326,531,327,524,208" onClick={this.handleClick}/>
            <area alt="grid"  data-value="wrong" shape="poly" coords="262,353,261,417,530,422,528,346" onClick={this.handleClick}/>
            <area alt="grid"  data-value="wrong" shape="poly" coords="265,440,261,507,523,511,536,438" onClick={this.handleClick}/>
            <area alt="grid"  data-value="wrong" shape="poly" coords="881,202,881,312,1016,305,1008,201" onClick={this.handleClick}/>
            <area alt="grid"  data-value="wrong" shape="poly" coords="880,337,881,434,1014,444,1016,332" onClick={this.handleClick}/>
            <area alt="grid"  data-value="wrong" shape="poly" coords="1055,198,1047,306,1144,305,1144,200" onClick={this.handleClick}/>
            <area alt="grid"  data-value="wrong" shape="poly" coords="884,461,883,558,1013,581,1012,468" onClick={this.handleClick}/>
            <area alt="grid"  data-value="wrong" shape="poly" coords="1045,467,1045,583,1148,603,1145,474" onClick={this.handleClick}/>
            <area alt="grid"  data-value="wrong" shape="poly" coords="872,591,871,643,988,641,985,590" onClick={this.handleClick}/>
            <area alt="grid"  data-inv="golden-ticket" data-value="correct" shape="poly" coords="1049,337,1049,447,1150,452,1146,335"
              onClick={({ target }) => this.handleClick(target.dataset.value, target.dataset.inv)}/>
          </map>
        </p>
        <p className="mission" > 
    Pick what candy to buy! Try to get the 'Golden Ticket'. 
        </p>
        <p className="message"> 
      Message: {this.state.message}
        </p>
      </section>
    );
  }

}

Candyshop.PropTypes = {
  message: PropTypes.string
};
