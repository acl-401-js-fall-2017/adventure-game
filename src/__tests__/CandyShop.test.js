import React from 'react';

import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Candyshop from '../components/Candyshop';

describe('CandyShop', () => {

  it('shows a candyshop', () => {
    const wrapper = shallow(<Candyshop handleInventory={() => {}} inventory={[]} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});