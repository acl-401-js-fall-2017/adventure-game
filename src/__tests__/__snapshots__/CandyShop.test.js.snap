// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CandyShop shows a candyshop 1`] = `
<section
  className="challenge"
>
  <p
    className="picture"
  >
    <img
      alt="Candy shop"
      src="wonkaStore.jpg"
      useMap="#shopMap"
    />
    <map
      id="shopMap"
      name="shopMap"
    >
      <area
        alt="grid"
        coords="53,194,52,320,220,324,220,208"
        data-value="wrong"
        onClick={[Function]}
        shape="poly"
      />
      <area
        alt="grid"
        coords="61,342,57,415,216,416,221,345"
        data-value="wrong"
        onClick={[Function]}
        shape="poly"
      />
      <area
        alt="grid"
        coords="56,444,52,518,205,516,220,443"
        data-value="wrong"
        onClick={[Function]}
        shape="poly"
      />
      <area
        alt="grid"
        coords="262,208,261,326,531,327,524,208"
        data-value="wrong"
        onClick={[Function]}
        shape="poly"
      />
      <area
        alt="grid"
        coords="262,353,261,417,530,422,528,346"
        data-value="wrong"
        onClick={[Function]}
        shape="poly"
      />
      <area
        alt="grid"
        coords="265,440,261,507,523,511,536,438"
        data-value="wrong"
        onClick={[Function]}
        shape="poly"
      />
      <area
        alt="grid"
        coords="881,202,881,312,1016,305,1008,201"
        data-value="wrong"
        onClick={[Function]}
        shape="poly"
      />
      <area
        alt="grid"
        coords="880,337,881,434,1014,444,1016,332"
        data-value="wrong"
        onClick={[Function]}
        shape="poly"
      />
      <area
        alt="grid"
        coords="1055,198,1047,306,1144,305,1144,200"
        data-value="wrong"
        onClick={[Function]}
        shape="poly"
      />
      <area
        alt="grid"
        coords="884,461,883,558,1013,581,1012,468"
        data-value="wrong"
        onClick={[Function]}
        shape="poly"
      />
      <area
        alt="grid"
        coords="1045,467,1045,583,1148,603,1145,474"
        data-value="wrong"
        onClick={[Function]}
        shape="poly"
      />
      <area
        alt="grid"
        coords="872,591,871,643,988,641,985,590"
        data-value="wrong"
        onClick={[Function]}
        shape="poly"
      />
      <area
        alt="grid"
        coords="1049,337,1049,447,1150,452,1146,335"
        data-inv="golden-ticket"
        data-value="correct"
        onClick={[Function]}
        shape="poly"
      />
    </map>
  </p>
  <p
    className="mission"
  >
    Pick what candy to buy! Try to get the 'Golden Ticket'.
  </p>
  <p
    className="message"
  >
    Message: 
  </p>
</section>
`;
